### PR TITLE
feat: use N-API 6 builds

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

Update the workflow and prebuild scripts to use N-API 6 builds for the nodeJS bindings.

## Links to any relevant issues

https://github.com/iotaledger/iota.rs/issues/1522

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Run the prebuild scripts locally

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/covector.yml [hunk 1]
```diff
@@ -92,16 +92,6 @@ jobs:
         # tests on Windows 2019, until we can figure out a way to fix the problem.
         # NOTE: Using Ubuntu 18.04 to provide glibc compatibility. (#588)
         os: [ubuntu-18.04, ubuntu-latest, macos-latest, windows-2019]
```
Comment (Thoralf-M) [inline .github/workflows/covector.yml]: Do we still need ubuntu-18.04 here?

Comment (rajivshah3) [reply .github/workflows/covector.yml]: I believe we're using 20.04 on `wallet.rs` bindings now so it's probably safe to do that here too

File: .github/workflows/covector.yml [hunk 2]
```diff
@@ -92,16 +92,6 @@ jobs:
...
-        node-version: ['14.x', '16.x', '18.x']
```
Comment (rajivshah3) [inline .github/workflows/covector.yml:95]: I'm on mobile so I can't check the default version that `actions/setup-node` uses, but I think the step on line 106 needs to be updated if we remove this. On `wallet.rs` we use 16.x (although tbh, any LTS version of Node should work as long as the scripts don't break)

Comment (Tuditi) [reply .github/workflows/covector.yml:95]: As far as I see, wallet.rs uses 14.x, but maybe it's worth updating that one to 16.x as well. I set it here to 16.x.

File: .github/workflows/covector.yml (hunk 1 above)
Comment (Tuditi) [reply .github/workflows/covector.yml]: It depends if we want to support older glibc versions. I updated it for now, but if users request it, we might need to roll back this change.

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**